### ISSUE TITLE
execute ignores basis_gates argument

### ISSUE THREAD

**Issue by chriseclectic** [Qiskit/qiskit-terra#2190]
### Information

- **Qiskit Terra version**: master
- **Python version**: 3.6
- **Operating system**: MacOS

### What is the current behavior?

It seems that the `basis_gates` kwarg of `qiskit.execute` is being ignored.

### Steps to reproduce the problem

If i run the following code
```python
import qiskit
from qiskit import *
from qiskit.providers.aer import *

qr = QuantumRegister(1)
cr = ClassicalRegister(1)
circ = QuantumCircuit(qr, cr)
circ.h(qr[0])
circ.x(qr[0])
circ.t(qr[0])
circ.measure(qr, cr)
job = qiskit.execute(circ, QasmSimulator(), basis_gates=['u3', 'cx'])
qobj = job.qobj()
qobj.experiments[0].instructions

# returns:
[QasmQobjInstruction(name='h', qubits=[0]),
 QasmQobjInstruction(name='x', qubits=[0]),
 QasmQobjInstruction(name='t', qubits=[0]),
 QasmQobjInstruction(memory=[0], name='measure', qubits=[0])]
```
So the circuit is not being unrolled to the specified basis gates.
### What is the expected behavior?

If I instead use the depreciated `qiskit.compile` i get the expected result:
```python
qobj = qiskit.compile(circ, QasmSimulator(), basis_gates=['u3', 'cx'])
qobj.experiments[0].instructions

# returns:
[QasmQobjInstruction(name='u3', params=[1.57079632679490, 0, 3.14159265358979], qubits=[0]),
 QasmQobjInstruction(name='u3', params=[3.14159265358979, 0, 3.14159265358979], qubits=[0]),
 QasmQobjInstruction(name='u3', params=[0, 0, 0.785398163397448], qubits=[0]),
 QasmQobjInstruction(memory=[0], name='measure', qubits=[0])]
```

### Suggested solutions

It looks like either the specified basis gates is not being passed to the unroller, or the unroller is not being called by the `execute` function.

**Comment by 1ucian0** [comment]
This is probably fixed by #2188. Can you check @chriseclectic?

**Comment by chriseclectic** [comment]
@1ucian0 no, doesn't seem to change anything. 

You might want to add some test cases where `backend != None`. Also it looks like all your tests go directly through the `transpiler.transpile` function, whereas the broken `execute` goes through the `compiler.transpile` function (the working `compile` function seems to go directly through `transpiler.transpile`)

**Comment by ajavadia** [comment]
This is fixed by #2166 
In that branch I get:
```
[QasmQobjInstruction(name='u3', params=[1.57079632679490, 0, 3.14159265358979], qubits=[0]),
 QasmQobjInstruction(name='u3', params=[3.14159265358979, 0, 3.14159265358979], qubits=[0]),
 QasmQobjInstruction(name='u3', params=[0, 0, 0.785398163397448], qubits=[0]),
 QasmQobjInstruction(memory=[0], name='measure', qubits=[0])]
```
